src/main/java/io/xcherry/core/persistence/schema/AppDatabaseSchema.java

```diff
@@ -0,0 +1,83 @@
+package io.xcherry.core.persistence.schema;
+
+import io.xcherry.core.exception.persistence.AppDatabaseSchemaNotMatchException;
+import io.xcherry.core.persistence.read_request.AppDatabaseReadRequest;
+import io.xcherry.core.persistence.read_request.AppDatabaseTableReadRequest;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import lombok.AccessLevel;
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor(access = AccessLevel.PRIVATE)
+public class AppDatabaseSchema {
+
+    /**
+     * table name: schema
+     */
+    private final Map<String, AppDatabaseTableSchema> tableSchemaMap;
+
+    /**
+     * Define an empty app database schema.
+     *
+     * @return  the created app database schema.
+     */
+    public static AppDatabaseSchema EMPTY() {
```

Comment:
Maybe `empty()` or make it a constant `public static AppDatabaseSchema  EMPTY;`

Reply:
There are other places using `EMPTY()`. I think we need to update for all of them. We can review and decide later.